src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -2854,16 +2854,68 @@ void C2_MacroAssembler::reconstruct_frame_pointer(Register rtmp) {
   }
 }
 
+// Implement selecting from two vectors using Neon instructions
 void C2_MacroAssembler::select_from_two_vectors_neon(FloatRegister dst, FloatRegister src1,
                                                      FloatRegister src2, FloatRegister index,
-                                                     FloatRegister tmp, BasicType bt, bool isQ) {
+                                                     FloatRegister tmp, unsigned vector_length_in_bytes) {
+  assert_different_registers(dst, src1, src2, tmp);
```

Comment:
It seems `dst` can be the same with either `src1`, `src2`, or `tmp` from following implementation instruction, right? Maybe we should assert more accurate for different cases, such as `src2 == src1 + 1` when `vector_length_in_bytes == 16`?